### PR TITLE
feat(ui): implement chunks

### LINKED ISSUE #57: dashboard chunk management

### PR DESCRIPTION
**Pull request by crherman7**
## Scope

[closes TICKET-57](https://github.com/crherman7/rechunk/issues/57)

### Major Updates:

1. **Database and Backend Enhancements**
   - Added Prisma seed logic to generate 10 sample chunks for testing and demonstration purposes.
   - Introduced a new `generateToken` utility function for RSA-signed chunk verification using `jsrsasign`.
   - Refactored chunk loader and action handlers to use cryptographic signing for improved security and integrity.

2. **New Routes and Components**
   - Added a new `routes/_dashboard.chunks.$chunkId` page to display detailed chunk information, including navigation between chunks.
   - Enhanced the `routes/_dashboard.chunks` page to list and display chunk metadata dynamically.
   - Introduced a `routes/api.v1.projects.$projectId.chunks.$chunkId` API endpoint for secure chunk retrieval.

3. **UI/UX Improvements**
   - Introduced new UI components using Radix UI:
     - `AlertDialog` for confirmation modals.
     - `ResizablePanelGroup` and `ScrollArea` for a responsive dashboard layout.
   - Improved visual styling of dashboard elements, including typography and spacing adjustments for better readability.
   - Replaced icons and buttons for a more cohesive design.

4. **Dependencies and Tooling**
   - Added dependencies for Radix UI components and `date-fns` for improved date handling.
   - Updated `pnpm-lock.yaml` to reflect new dependencies.

5. **General Refactoring**
   - Simplified custom hooks `useChunk` and `useChunkIds` for easier retrieval of chunk data from loaders.
   - Refined layouts in `routes/_dashboard.settings` for better consistency and alignment.

## Screenshots

<details>
  <summary>browser</summary>

  

https://github.com/user-attachments/assets/589a2fa9-0b2d-493a-8e32-bb956a8cb3f5
</details>

## How to Test

- `pnpm install`
- `cd apps/www`
- `cp .env.example` `.env`
- `pnpm seed`
- `pnpm dev`
- Click "Get Started"
- Login with id/write key: `cm38wkhur0000yseh1beq2hpx`/`write-2b05d863-6100-458b-b527-bc5a9d0b2061`
- Verify you get redirected to `/chunks` route
- Verify a list of chunks
- Click on a chunk
- Verify redirect to the specific chunk route
- Click delete icon
- Verify deletion of chunk
- Click Next and Previous icon
- Verify appropriate redirects

## Emoji Guide

**For reviewers: Emojis can be added to comments to call out blocking versus non-blocking feedback.**

E.g: Praise, minor suggestions, or clarifying questions that don’t block merging the PR.

> 🟢 Nice refactor!

> 🟡 Why was the default value removed?

E.g: Blocking feedback must be addressed before merging.

> 🔴 This change will break something important

| | | |
| --- | --- | --- |
| Blocking | 🔴 ❌ 🚨 | RED |
| Non-blocking | 🟡 💡 🤔 💭 | Yellow, thinking, etc |
| Praise | 🟢 💚 😍 👍 🙌 | Green, hearts, positive emojis, etc |

### CODE REVIEW

File: apps/www/app/routes/_dashboard.chunks.$chunkId.tsx [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+                    size="icon"
+                    className="rounded-lg hover:bg-muted"
+                    aria-label="Chunks">
+                    <ArrowLeftIcon />
+                  </Button>
+                </TooltipTrigger>
+                <TooltipContent side="top" sideOffset={5}>
+                  Previous
+                </TooltipContent>
+              </Tooltip>
```
Comment (crherman7) [inline apps/www/app/routes/_dashboard.chunks.$chunkId.tsx]: can probably be moved to it's own component

File: apps/www/app/routes/_dashboard.chunks.$chunkId.tsx [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+                    <ArrowRightIcon />
+                  </Button>
+                </TooltipTrigger>
+                <TooltipContent side="top" sideOffset={5}>
+                  Next
+                </TooltipContent>
+              </Tooltip>
+            </div>
+          </div>
+        </div>
```
Comment (crherman7) [inline apps/www/app/routes/_dashboard.chunks.$chunkId.tsx]: can probably be moved to it's own component

File: apps/www/app/routes/_dashboard.chunks.$chunkId.tsx (hunk 1 above)
Comment (crherman7) [inline apps/www/app/routes/_dashboard.chunks.$chunkId.tsx]: can probably be moved to it's own component

File: apps/www/app/routes/_dashboard.chunks.$chunkId.tsx [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+    if (!chunkIds || !chunk?.id) return;
+
+    const currentIndex = chunkIds.findIndex(it => it === chunk.id);
+
+    if (currentIndex === -1) return;
+
+    const previousIndex =
+      (currentIndex - 1 + chunkIds.length) % chunkIds.length;
+    navigate(`/chunks/${chunkIds[previousIndex]}`);
+  }
```
Comment (crherman7) [inline apps/www/app/routes/_dashboard.chunks.$chunkId.tsx]: moved to it's abstracted component

File: apps/www/app/routes/_dashboard.chunks.$chunkId.tsx [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+  function onNext() {
+    if (!chunkIds || !chunk?.id) return;
+
+    const currentIndex = chunkIds.findIndex(it => it === chunk.id);
+
+    if (currentIndex === -1) return;
+
+    const nextIndex = (currentIndex + 1) % chunkIds.length;
+    navigate(`/chunks/${chunkIds[nextIndex]}`);
+  }
```
Comment (crherman7) [inline apps/www/app/routes/_dashboard.chunks.$chunkId.tsx]: moved to it's abstracted component

File: apps/www/app/routes/_dashboard.chunks.tsx [hunk 5]
```diff
@@ -1,7 +1,87 @@
...
+                  </p>
+                  <p className="text-xs text-slate-400 font-medium font-mono">
+                    {it.projectId}
+                  </p>
+                  <div className="bg-slate-50 p-2 rounded-lg">
+                    <p className="text-wrap line-clamp-3 text-[10px] text-slate-400 font-mono font-extralight break-all">
+                      {it.data}
+                    </p>
+                  </div>
+                </Link>
```
Comment (crherman7) [inline apps/www/app/routes/_dashboard.chunks.tsx]: moved to it's abstracted component

File: apps/www/app/routes/_dashboard.chunks.$chunkId.tsx (hunk 1 above)
Comment (devs9) [reply apps/www/app/routes/_dashboard.chunks.$chunkId.tsx]: good idea 👍

File: apps/www/app/routes/_dashboard.chunks.$chunkId.tsx (hunk 3 above)
Comment (devs9) [reply apps/www/app/routes/_dashboard.chunks.$chunkId.tsx]: I like that 👍

File: apps/www/app/routes/_dashboard.chunks.tsx (hunk 5 above)
Comment (devs9) [reply apps/www/app/routes/_dashboard.chunks.tsx]: good idea 👍

Comment (devs9) [reply apps/www/app/routes/_dashboard.chunks.tsx]: 🤔  also, what do you think about moving the format(`"eeee, MMMM do, yyyy 'at' hh:mm a"`) into a constant? it repeats in several places

File: apps/www/app/routes/_dashboard.chunks.$chunkId.tsx (hunk 1 above)
Comment (crherman7) [reply apps/www/app/routes/_dashboard.chunks.$chunkId.tsx]: updated

File: apps/www/app/routes/_dashboard.chunks.$chunkId.tsx (hunk 2 above)
Comment (crherman7) [reply apps/www/app/routes/_dashboard.chunks.$chunkId.tsx]: updated

File: apps/www/app/routes/_dashboard.chunks.$chunkId.tsx (hunk 1 above)
Comment (crherman7) [reply apps/www/app/routes/_dashboard.chunks.$chunkId.tsx]: updated

File: apps/www/app/routes/_dashboard.chunks.$chunkId.tsx (hunk 3 above)
Comment (crherman7) [reply apps/www/app/routes/_dashboard.chunks.$chunkId.tsx]: updated

File: apps/www/app/routes/_dashboard.chunks.$chunkId.tsx (hunk 4 above)
Comment (crherman7) [reply apps/www/app/routes/_dashboard.chunks.$chunkId.tsx]: updated

File: apps/www/app/routes/_dashboard.chunks.tsx (hunk 5 above)
Comment (crherman7) [reply apps/www/app/routes/_dashboard.chunks.tsx]: updated

File: apps/www/app/components/ChunkItem.tsx [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+      <div className="mb-2 flex w-full flex-row items-center justify-between">
+        <p className="text-sm font-bold">{id}</p>
+        <p className="text-xs font-normal">
+          {formatDistanceToNow(new Date(createdAt), {
+            addSuffix: true,
+          })}
+        </p>
+      </div>
+      <p className="text-xs text-muted-foreground">
+        {format(new Date(updatedAt), "eeee, MMMM do, yyyy 'at' hh:mm a")}
```
Comment (devs9) [inline apps/www/app/components/ChunkItem.tsx]:
```suggestion
        {format(new Date(updatedAt), DATE_FORMAT)}
```

**Review by devs9 — APPROVED** [review]
Awesome job 💚 the chunks page looks great 👍

File: apps/www/app/components/ChunkItem.tsx (hunk 6 above)
Comment (crherman7) [reply apps/www/app/components/ChunkItem.tsx]: ahh yes! let me update this

**Review by devs9 — APPROVED** [review]

**Pull request merged**